### PR TITLE
Refresh token rotation documentation

### PR DESCRIPTION
**Pull request by nabokihms**
Documentation for https://github.com/dexidp/dex/pull/1846

### CODE REVIEW

File: content/docs/id-tokens.md [hunk 1]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
...
+      
+__NOTE__: `disableRotation` and `reuseInterval` options help effectively deal with network lags, concurrent requests, and so on in tradeoff for security. Use them with caution.
+  
+
 [jwt-io]: https://jwt.io/
 [standard-claims]: https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
 [kubernetes]: http://kubernetes.io/docs/admin/authentication/#openid-connect-tokens
 [aws-sts]: https://docs.aws.amazon.com/STS/latest/APIReference/Welcome.html
 [using-dex]: using-dex.md
+[connectors]: connectors/_index.md
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:87]: minor nit: could we alphabetize these links in the footnotes section?

File: content/docs/id-tokens.md [hunk 2]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
 
 For details on how to request or validate an ID Token, see [“_Writing apps that use Dex_.”][using-dex]
 
+## Refresh tokens
+Refresh tokens are credentials used to obtain access tokens.  Refresh tokens are issued to the client by the authorization server and are used to obtain
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:47]: We only need one space after a period.

```suggestion
Refresh tokens are credentials used to obtain access tokens. Refresh tokens are issued to the client by the authorization server and are used to obtain 
```

File: content/docs/id-tokens.md [hunk 3]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
 
 For details on how to request or validate an ID Token, see [“_Writing apps that use Dex_.”][using-dex]
 
+## Refresh tokens
+Refresh tokens are credentials used to obtain access tokens.  Refresh tokens are issued to the client by the authorization server and are used to obtain 
+a new id token when the current id token becomes invalid or expires. Issuing a refresh token is optional and provided by passing `offline_access` scope to Dex server.
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:48]:
```suggestion
a new id token when the current id token becomes invalid or expires. Issuing a refresh token is optional and is provided by passing `offline_access` scope to Dex server.
```

File: content/docs/id-tokens.md (hunk 1 above)
Comment (nate-double-u) [inline content/docs/id-tokens.md:87]: We only need `connectors/` here I beleive, no need for the `_index.md` as the index files get resolved to the folder. Same for `[using-dex]` above.

**Comment by nate-double-u** [comment]
Deploy preview: https://deploy-preview-96--dexidp.netlify.app/docs/id-tokens/

File: content/docs/id-tokens.md (hunk 1 above)
Comment (chalin) [reply content/docs/id-tokens.md:87]: Right: drop `/_index.md` and, in general,`.md` suffixes.

File: content/docs/id-tokens.md [hunk 4]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
...
+{
+ "access_token": "eyJhbGciOiJSUzI1N...",
+ "token_type": "Bearer",
+ "refresh_token": "lxzzsvasxho5exvwkfa5zhefl",
+ "expires_in": 3600,
+ "id_token": "eyJhbGciO..."
+}
+```
+
+__NOTE__: For every refresh of id token, Dex issues a new refresh token. This security measure is called refresh token rotation
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:63]:
```suggestion
__NOTE__: For every refresh of an id token, Dex issues a new refresh token. This security measure is called _refresh token rotation_
```

File: content/docs/id-tokens.md [hunk 5]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
...
+and prevents stealing it. The idea is described in detail in the corresponding [RFC][rfc6819-5.2.2.3].
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:64]:
```suggestion
and prevents someone stealing it. The idea is described in detail in the corresponding [RFC][rfc6819-5.2.2.3].
```

File: content/docs/id-tokens.md [hunk 6]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
...
+
+## Expiration and rotation settings
+
+Dex has a section in the config file where you can specify expiration and rotation settings for id tokens and refresh tokens.
+__NOTE__: All duration options should be set in the format: number + time unit (s, m, h), e.g., `10m`.
+
+* `expire` - section for various expiration settings, including token settings:
+  * `idTokens` - the lifetime of id_token. It is preferable to use short-lived id tokens.
+  * `refreshTokens` - section for various refresh token settings:
+    * `validForIfNotUsed` - invalidate a refresh token if it is not used for specified amount of time.
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:74]:
```suggestion
     * `validForIfNotUsed` - invalidate a refresh token if it is not used for a specified amount of time.
```

File: content/docs/id-tokens.md [hunk 7]
```diff
@@ -41,8 +43,46 @@ Because these tokens are signed by Dex and [contain standard-based claims][stand
...
+    * `absoluteLifetime` - a stricter variant of the previous option, absolute lifetime of a refresh token. It forces users to reauthenticate and obtain a new refresh token.
+    * `disableRotation` - completely disables every-request rotation. The user will also have to specify one of the previous refresh token options to keep refresh tokens secure by toggling this.
```
Comment (nate-double-u) [inline content/docs/id-tokens.md:76]:
```suggestion
    * `disableRotation` - completely disables every-request rotation. The user will also have to specify one of the previous refresh token options to keep refresh tokens secure when toggling this.
```

**Comment by nate-double-u** [comment]
@nabokihms, thanks for this. Just a couple edits in-line, but otherwise it looks good to me.

/lgtm
/hold for https://github.com/dexidp/dex/pull/1846

File: content/docs/id-tokens.md (hunk 1 above)
Comment (nabokihms) [reply content/docs/id-tokens.md]: Yeah, no problem.

**Review by sagikazarmark — APPROVED** [review]

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**